### PR TITLE
python311Packages.pynndescent: fix build

### LINKED ISSUE #284668: Build failure: pythonPackages.pynndescent
### Steps To Reproduce

1. build pythonPackages.pynndescent

### Build log

```
warning: Git tree '/home/dennis/repos/RoutineAI' is dirty
evaluating derivation 'git+file:///home/dennis/repos/RoutineAI#devShells.x86_64-linux.default'direnv: ([direnv export zsh]) is taking a while to execute. Use CTRL-C to give up.
error: builder for '/nix/store/zq2maq7q3jns86v8nsrxqqc7szvjdx1d-python3.11-pynndescent-0.5.11.drv' failed with exit code 1;
       last 10 log lines:
       > unpacking source archive /nix/store/bl0lk5m4i61libjq3zs8f1v1bsdfpd2z-pynndescent-0.5.11.tar.gz
       > source root is pynndescent-0.5.11
       > setting SOURCE_DATE_EPOCH to timestamp 1700506216 of file pynndescent-0.5.11/setup.cfg
       > Running phase: patchPhase
       > applying patch /nix/store/ddg3fz50r4fa3fzrhbq05qv062cz6x2b-86e0d716a3a4d5f4e6a0a3c2952f6fe339524e96.patch
       > patching file pynndescent/tests/test_distances.py
       > Reversed (or previously applied) patch detected!  Assume -R? [n]
       > Apply anyway? [n]
       > Skipping patch.
       > 5 out of 5 hunks ignored -- saving rejects to file pynndescent/tests/test_distances.py.rej
       For full logs, run 'nix log /nix/store/zq2maq7q3jns86v8nsrxqqc7szvjdx1d-python3.11-pynndescent-0.5.11.drv'.
error: 1 dependencies of derivation '/nix/store/1vr6s9f0y6kzp5gkgn534ni7ms7mml6i-python3-3.11.7-env.drv' failed to build
error: 1 dependencies of derivation '/nix/store/5jy0qqizgraywqm7pcyb3l6in47mramn-nix-shell-env.drv' failed to build
```

### Notify maintainers

@Mic92 

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.13, NixOS, 24.05 (Uakari), 24.05.20240125.ae5c332`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by dotlambda**
## Description of changes
fixes https://github.com/NixOS/nixpkgs/issues/284668

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by fabaff — COMMENTED** [review]
Line 4: `fetchpatch` could also be removed.

**Review by DerDennisOP — APPROVED** [review]

File: pkgs/development/python-modules/pynndescent/default.nix [hunk 1]
```diff
@@ -46,16 +42,6 @@ buildPythonPackage rec {
     pytestCheckHook
   ];
 
-  disabledTests = [
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pynndescent/default.nix:49]: A whole bunch of tests fail on darwin:

```
=========================== short test summary info ============================
FAILED cluster/tests/test_mean_shift.py::test_parallel[float64] - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED decomposition/tests/test_dict_learning.py::test_sparse_encode_shapes_omp - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED decomposition/tests/test_dict_learning.py::test_dict_learning_reconstruction_parallel - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED decomposition/tests/test_dict_learning.py::test_dict_learning_lassocd_readonly_data - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED decomposition/tests/test_dict_learning.py::test_sparse_coder_parallel_mmap - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED decomposition/tests/test_dict_learning.py::test_cd_work_on_joblib_memmapped_data - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED ensemble/tests/test_bagging.py::test_parallel_classification - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED ensemble/tests/test_bagging.py::test_parallel_regression - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED ensemble/tests/test_bagging.py::test_estimator - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED ensemble/tests/test_forest.py::test_backend_respected - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED ensemble/tests/test_voting.py::test_parallel_fit[42] - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED feature_selection/tests/test_rfe.py::test_rfe_cv_n_jobs[42] - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED gaussian_process/tests/test_gpc.py::test_multi_class_n_jobs[kernel0] - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED gaussian_process/tests/test_gpc.py::test_multi_class_n_jobs[kernel1] - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED gaussian_process/tests/test_gpc.py::test_multi_class_n_jobs[kernel2] - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED gaussian_process/tests/test_gpc.py::test_multi_class_n_jobs[kernel3] - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED inspection/tests/test_permutation_importance.py::test_permutation_importance_correlated_feature_regression[0.5-2] - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED inspection/tests/test_permutation_importance.py::test_permutation_importance_correlated_feature_regression[1.0-2] - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED inspection/tests/test_permutation_importance.py::test_robustness_to_high_cardinality_noisy_feature[0.5-2] - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED inspection/tests/test_permutation_importance.py::test_robustness_to_high_cardinality_noisy_feature[1.0-2] - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED inspection/tests/test_permutation_importance.py::test_permutation_importance_equivalence_sequential_parallel[500] - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED inspection/tests/test_permutation_importance.py::test_permutation_importance_equivalence_sequential_parallel[1.0] - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED inspection/tests/test_permutation_importance.py::test_permutation_importance_large_memmaped_data[array] - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED linear_model/tests/test_coordinate_descent.py::test_linear_models_cv_fit_with_loky[ElasticNetCV] - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED linear_model/tests/test_coordinate_descent.py::test_linear_models_cv_fit_with_loky[LassoCV] - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED linear_model/tests/test_sgd.py::test_multi_core_gridsearch_and_early_stopping - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED linear_model/tests/test_theil_sen.py::test_theil_sen_parallel - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED manifold/tests/test_mds.py::test_MDS - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED metrics/tests/test_classification.py::test_classification_metric_division_by_zero_nan_validaton[scoring0] - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED metrics/tests/test_classification.py::test_classification_metric_division_by_zero_nan_validaton[scoring1] - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED metrics/tests/test_classification.py::test_classification_metric_division_by_zero_nan_validaton[scoring2] - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED metrics/tests/test_classification.py::test_classification_metric_division_by_zero_nan_validaton[scoring3] - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED metrics/tests/test_pairwise.py::test_sparse_manhattan_readonly_dataset[csr_matrix] - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED metrics/tests/test_pairwise.py::test_sparse_manhattan_readonly_dataset[csr_array] - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED model_selection/tests/test_search.py::test_random_search_bad_cv - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED model_selection/tests/test_validation.py::test_cross_validate_many_jobs - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED model_selection/tests/test_validation.py::test_cross_validate_return_indices[42] - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED neighbors/tests/test_kd_tree.py::test_kdtree_picklable_with_joblib[KDTree64] - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED neighbors/tests/test_kd_tree.py::test_kdtree_picklable_with_joblib[KDTree32] - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED neighbors/tests/test_neighbors.py::test_knn_forcing_backend[ball_tree-loky] - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED neighbors/tests/test_neighbors.py::test_knn_forcing_backend[kd_tree-loky] - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED neighbors/tests/test_neighbors.py::test_knn_forcing_backend[auto-loky] - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED tests/test_calibration.py::test_parallel_execution[True-sigmoid] - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED tests/test_calibration.py::test_parallel_execution[True-isotonic] - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED tests/test_calibration.py::test_parallel_execution[False-sigmoid] - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED tests/test_calibration.py::test_parallel_execution[False-isotonic] - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED tests/test_config.py::test_config_threadsafe_joblib[loky] - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED tests/test_multioutput.py::test_multi_output_classification_partial_fit_parallelism - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED tests/test_pipeline.py::test_feature_union_parallel - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
FAILED utils/tests/test_parallel.py::test_configuration_passes_through_to_joblib[loky-2] - joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed...
= 50 failed, 28391 passed, 3719 skipped, 88 xfailed, 45 xpassed, 3072 warnings in 977.71s (0:16:17) =
```

Comment (dotlambda) [reply pkgs/development/python-modules/pynndescent/default.nix:49]: Let's merge anyway. These can be fixed later.

Comment (SuperSandro2000) [reply pkgs/development/python-modules/pynndescent/default.nix:49]: I mean we could have marked it broken because to many test failures 😅

**Pull request merged**